### PR TITLE
Tests: price floors

### PR DESCRIPTION
**Pull request by mtuchkova**
Tests for #1619

### CODE REVIEW

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        floorsProvider.setResponse(bidRequest.site.publisher.id, floorsResponse)
+
+        when: "PBS processes auction request"
+        floorsPbsService.sendAuctionRequest(bidRequest)
+
+        then: "Bidder request should contain bidFloor, bidFloorCur from floors provider"
+        def bidderRequest = bidder.getBidderRequest(bidRequest.id)
+        verifyAll(bidderRequest) {
+            imp[0]?.bidFloor == floorValue
+            imp[0]?.bidFloorCur == floorsResponse.data.modelGroups[0].currency
```
Comment (hupaloo) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:46]: No point in using null-safe operator here. If imp is null, so then imp[0] will return NPE exception. Check pls similar places.

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+            ext.prebid.floors.floorMinCur = USD
+        }
+
+        and: "Account with enabled fetch, fetch.url in the DB"
+        def account = getAccountWithEnabledFetch(bidRequest.site.publisher.id)
+        accountDao.save(account)
+
+        and: "Set Floors Provider response with a currency different from the floorMinCur"
+        def floorsResponse = PriceFloorRules.priceFloorRules.tap {
+            data.modelGroups[0].values = [(rule): randomFloorValue]
```
Comment (bretg) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsCurrencySpec.groovy:146]: Is this really a random function? Seems like this ought to be based on the floorMin above?

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy [hunk 3]
```diff
@@ -0,0 +1,1189 @@
...
+        assert metrics[metric] == 1
+
+        and: "PBS floors validation failure should not reject the entire auction"
+        assert !response.seatbid.isEmpty()
+    }
+
+    def "PBS should validate fetch.period-sec from account config"() {
+        given: "Pbs with PF configuration with minMaxAgeSec, maxAgeSec"
+        def minMaxAgeSec = PBSUtils.randomNumber
+        def maxAgeSec = PBSUtils.randomNumber + 10
```
Comment (bretg) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy:159]: This doesn't seem right? If the random number is different on each line, then minMaxAgeSec might be larger or smaller than the maxAge. Seems like we want it consistent within the test. Like "def maxAgeSec=minMaxAgeSec+10"

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy [hunk 4]
```diff
@@ -0,0 +1,1189 @@
...
+        assert metrics[metric] == 1
+
+        and: "PBS floors validation failure should not reject the entire auction"
+        assert !response.seatbid?.isEmpty()
+    }
+
+    def "PBS should validate fetch.timeout-ms from account config"() {
+        given: "Pbs with PF configuration with timeoutMs, minTimeoutMs, maxTimeoutMs"
+        def minTimeoutMs = PBSUtils.randomNumber
+        def maxTimeoutMs = PBSUtils.randomNumber + 10
```
Comment (bretg) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsFetchingSpec.groovy:275]: same comment as above - if randomNumber is generated separately, max could be less than min here.

File: src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsSignalingSpec.groovy [hunk 5]
```diff
@@ -0,0 +1,532 @@
...
+        assert bidderRequest.imp[0].ext.prebid.floors.floorRuleValue == floorsProviderFloorValue
+        assert bidderRequest.imp[0].ext.prebid.floors.floorValue == floorsProviderFloorValue
+
+        where:
+        pbsConfigBidAdjustmentFlag | requestBidAdjustmentFlag | accountBidAdjustmentFlag
+        false                      | false                    | null
+        true                       | null                     | false
+    }
+
+    def "PBS placeholder"() {
```
Comment (bretg) [inline src/test/groovy/org/prebid/server/functional/tests/pricefloors/PriceFloorsSignalingSpec.groovy:441]: ?

File: src/test/groovy/org/prebid/server/functional/model/db/StoredRequest.groovy [hunk 6]
```diff
@@ -34,8 +34,10 @@ class StoredRequest {
         new StoredRequest(reqid: ampRequest.tagId, accountId: ampRequest.account, requestData: bidRequest)
     }
 
-    static StoredRequest getDbStoredRequest(BidRequest bidRequest, BidRequest storedRequest) {
-        new StoredRequest(reqid: bidRequest.ext.prebid.storedRequest.id, accountId: bidRequest.site.publisher.id,
+    static StoredRequest getDbStoredRequest(BidRequest bidRequest,
+                                            String accountId = bidRequest.site.publisher.id,
```
Comment (Net-burst) [inline src/test/groovy/org/prebid/server/functional/model/db/StoredRequest.groovy:38]: 1) Arguments with default values should go last unless really need to lessen possible confusion.
2) Please use null-safe reference.

**Review by SerhiiNahornyi — APPROVED** [review]

**Pull request merged**